doc/source/how-to-upgrade-to-flower-next.rst

```diff
@@ -0,0 +1,169 @@
+
+    # Flower 1.8
+    def client_fn(cid: str):
+        return flwr.client.FlowerClient().to_client() 
+    
+    app = flwr.client.ClientApp(
+       client_fn=client_fn,
+    )
+
+    # Flower 1.7
+    if __name__ == "__main__":
+        flwr.client.start_client(
+           server_address="127.0.0.1:8080",
+           client=flwr.client.FlowerClient().to_client(),
+        )
+
+``ServerApp``
+~~~~~~~~~~~~~
+- Use ``flwr.server.ServerApp(config, strategy)`` instead of ``flwr.server.start_server(server_address, config, strategy)``.
+  Here's an example:
+
+.. code-block:: python
+
+    # Flower 1.8
+    app = flwr.server.ServerApp(
+        config=config,
+        strategy=strategy,
+    )
+
+    # Flower 1.7
+    if __name__ == "__main__":
+        flwr.server.start_server(
+            server_address="0.0.0.0:8080",
+            config=config,
+            strategy=strategy,
+        )
+
+Deployment
+~~~~~~~~~~
+Run the ``SuperLink`` before running ``ServerApp`` and ``SuperNode`` instead of executing `client.py` and
+`server.py` as Python scripts. Here's an example:
+
+.. code-block:: bash
+    
+    # Start a Superlink
+    $ flower-superlink --insecure
+
+    # In a new terminal window, start a long-running SuperNode
+    $ flower-client-app client:app --insecure
+
+    # In another terminal window, start a long-running SuperNode (at least 2 SuperNodes are required)
+    $ flower-client-app client:app --insecure
+
+    # In another terminal window, run the apps
+    $ flower-server-app server:app --insecure
+
+Simulation
+~~~~~~~~~~
+- Use ``flwr.client.ClientApp()`` and ``flwr.server.ServerApp()`` instead of ``flwr.simulation.start_simulation()``.
+  Here's an example:
+
+.. code-block:: python
+
+    # Flower 1.8
+    def client_fn(cid: str):
+        return flwr.client.FlowerClient().to_client() 
+    
+    client = flwr.client.ClientApp(
+       client_fn=client_fn,
+    )
+
+    server = flwr.server.ServerApp(
+        config=config,
+        strategy=strategy,
+    )
+
+    # Flower 1.7
+    if __name__ == "__main__":
+        flwr.simulation.start_simulation(
+            ...
+        )
+
+- Run ``flower-simulation`` in CLI and point to the ``server``/``client`` object in the code instead of
+  executing the Python script. Here's an example (assuming the ``server`` and ``client`` are in a ``sim.py`` file):
+
+.. code-block:: bash
+
+    # Flower 1.8
+    $ flower-simulation --client-app=sim:client --server-app=sim:server --num-supernodes=100
+
+    # Flower 1.7
+    $ python sim.py
+
+- Change default resources for each ``ClientApp`` using the ``--backend-config`` argument instead of custom arguments
+  parsed by ``argparse``. Here's an example:
+
+.. code-block:: bash
+
+    # Flower 1.8
+    $ flower-simulation --client-app=sim:client --server-app=sim:server --num-supernodes=100 \
+        --backend-config='{"client_resources": {"num_cpus":2, "num_gpus":0.25}}'
+
+    # Flower 1.7
+    $ python sim.py --num_cpus=2 --num_gpus=0.25
```

Comment:
```suggestion
    # specify `client_resources` when launching `flwr.simulation.start_simulation()`
```

Reply:
I think it would be better to replace it because the `python sim.py --num_cpus=2 ...` it only works if your `sim.py` is designed to accept those (and pass them to `start_simulation()`. Maybe there is a better way of conveying this idea than what i propose above?

Reply:
Good point. I've revised the sentence to clarify along the lines that the Flower Next approach "passes command line arguments" and the old approach as "setting the `client_resources` argument in `start_simulation()`".